### DISCUSSION TITLE
Where is the app permission grant stored?

### DISCUSSION THREAD

**Discussion by robertcaretta** [pnp/powershell#2337] [Q&A]
Where is the app permission grant stored when using this cmdlet?
Grant-PnPAzureADAppSitePermission

I was able to successfully run this and grant the Azure AD app write access to the site. That is fine.  However when I review these SPO screens I do not see anything registered...

Site App Permissions
/_layouts/15/appprincipals.aspx?Scope=Web

Site Collection App Permissions
/_layouts/15/appprincipals.aspx

**Comment by LeonArmston** [comment]
@robertcaretta 

I've had a look and I cant seem to spot them in the UI anywhere.

The only way I've been able to see them is through running the PowerShell [Get-PnPAzureADAppSitePermission](https://github.com/pnp/powershell/blob/dev/documentation/Get-PnPAzureADAppSitePermission.md). The permissions seem to be stored in MS Graph somewhere.

**Comment by robertcaretta** [comment]
Thanks for checking. That's what I observed as well.
